tests/test_fixtures.py

```diff
@@ -233,12 +232,14 @@ def output(self):
     def wait_for_completion(self, timeout=120):
         self._process.wait(timeout)
 
-    def wait_for_provision(self, timeout=60):
-        while True:
-            info = self.get_info()
-            if 'Provisioned on server: yes' in info:
-                break
+    def wait_for_provision(self, timeout=20):
```

Comment:
What this 'timeout' param is for ?

Frankly speaking, I am getting annoyed by nondeterministic behavior of aktualizr-info, it doesn't yield the desired result even after a device is provisioned, looks like it happens when it tries to access the DB when aktualizr writes some data to it, just guessing here...

Reply:
> I am getting annoyed by nondeterministic behavior of aktualizr-info, it doesn't yield the desired result even after a device is provisioned, looks like it happens when it tries to access the DB when aktualizr writes some data to it, just guessing here...

Yeah, that has sporadically cropped up from time to time, and the general suspicion is the database access issue you mention. Perhaps aktualizr-info needs a built-in retry?

Reply:
> What this 'timeout' param is for ?

The method is a `wait_for_xxx`, so it's here to prevent an infinite block if it never provisions.

Reply:
Yes, you're right. Just saw the typo...